### PR TITLE
Build fix with MinGW GCC 9.2.0

### LINKED ISSUE #12065: Community triplets not working because of toolchain detection mechanism
Trying to use `*-mingw` triplets gives me an `Unable to determine toolchain to use for triplet` error. There are no references to community triplets in the documentation either. Am I missing something here?

### PR DESCRIPTION
**Pull request by cristianadam**
**Describe the pull request**

This PR will allow building vcpkg itself with MinGW GCC 9.2.0. It fixes some warnings and because warnings are treated as errors, the build would fail.

It also fixes the usage of MinGW community tripplets.

Fixes: #12065

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

File: toolsrc/src/vcpkg/build.cpp [hunk 1]
```diff
@@ -450,7 +450,7 @@ namespace vcpkg::Build
         std::ofstream out_file(stdoutlog.native().c_str(), std::ios::out | std::ios::binary | std::ios::trunc);
         Checks::check_exit(VCPKG_LINE_INFO, out_file, "Failed to open '%s' for writing", stdoutlog.u8string());
         std::string compiler_hash;
-        const int return_code = System::cmd_execute_and_stream_lines(
+        /*const int return_code = */System::cmd_execute_and_stream_lines(
```
Comment (strega-nil) [inline toolsrc/src/vcpkg/build.cpp:453]: This seems incorrect.

**Review by strega-nil — APPROVED** [review]
Can you merge to latest master? This seems like something we definitely want.

**Review by ras0219-msft — APPROVED** [review]

**Comment by cristianadam** [comment]
> Can you merge to latest master? This seems like something we definitely want.

I've rebased on master.

**Pull request merged**